she/verify.py

```
@@ -0,0 +1,239 @@
+def apply_check_results(
+    plan: VerificationPlan,
+    results: Mapping[str, Mapping[str, Any]],
+) -> VerificationPlan:
+    """Record outcomes onto an existing plan. Cannot drop required gates."""
+    by_gate = {c.gate: c for c in plan.checks}
+    updated: list[CheckSpec] = []
+    for gate, spec in by_gate.items():
+        payload = results.get(gate) or {}
+        outcome = str(payload.get("outcome") or spec.outcome)
+        if outcome not in CHECK_OUTCOMES:
+            raise VerificationError(f"unknown outcome for {gate}: {outcome!r}")
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)